pandas/tseries/frequencies.py

```diff
@@ -239,17 +239,18 @@ def get_freq(self) -> Optional[str]:
         if not self.is_monotonic or not self.index._is_unique:
             return None
 
-        delta = self.deltas[0]
-        if _is_multiple(delta, _ONE_DAY):
+        delta = min(self.deltas)
+        if delta and _is_multiple(delta, _ONE_DAY):
             return self._infer_daily_rule()
 
         # Business hourly, maybe. 17: one day / 65: one weekend
         if self.hour_deltas in ([1, 17], [1, 65], [1, 17, 65]):
             return "BH"
+
         # Possibly intraday frequency.  Here we use the
         # original .asi8 values as the modified values
         # will not work around DST transitions.  See #8772
-        elif not self.is_unique_asi8:
+        if not self.is_unique_asi8:
```

Comment:
unnecessary diff tho ofc doesn't change the logic

Reply:
While the code does the same with or without the diff, the new version is more readable/understandable in terms of logic (as the first if has a return, it cuts the flow of the function and the elif is just equivalent to a new if ... hmmm, not sure to make my point clear ;-) )

Reply:
yeah that's fine